backend-lib/src/main/rust/lib.rs

```diff
@@ -145,16 +145,42 @@ fn account_id_from_jni<'local, P: Parameters>(
     }
 }
 
+/// Initializes global Rust state, such as the logging infrastructure and threadpools.
+///
+/// When `log_level` defines how the Rust layer logs its events. These values are supported:
+/// `trace` - Logs very low priority, often extremely verbose, information
+/// `debug` - Logs lower priority information
+/// `off` - The logs are completely disabled
+///
+/// # Panics
+///
+/// This method panics if called more than once.
 #[no_mangle]
 pub extern "C" fn Java_cash_z_ecc_android_sdk_internal_jni_RustBackend_initOnLoad<'local>(
     _env: JNIEnv<'local>,
     _: JClass<'local>,
+    _log_level: JString<'local>,
 ) {
     // Set up the Android tracing layer.
     #[cfg(target_os = "android")]
-    let android_layer = paranoid_android::layer("cash.z.rust.logs")
-        .with_ansi(false)
-        .with_filter(tracing_subscriber::filter::LevelFilter::INFO);
+    let android_layer = {
+        let log_filter = _env
+            .get_string(_log_level)
```

Comment:
@str4d This proposed implementation produces a mismatched type error while building it.
```
Compiling zcash-android-wallet-sdk v0.0.4 (/Users/honza/Projects/zcash-android-wallet-sdk/backend-lib)
error[E0308]: mismatched types
    --> src/main/rust/lib.rs:168:25
     |
168  |             .get_string(_log_level)
     |              ---------- ^^^^^^^^^^ expected `&JString<'_>`, found `JString<'_>`
     |              |
     |              arguments to this method are incorrect
```
See more details in the failed build CI job. Thank you for taking a look at it.